pkg/azschedulerextender/main.go

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"encoding/json"
+	"math/rand"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog/v2"
+	schedulerapi "k8s.io/kube-scheduler/extender/v1"
+	"time"
+)
+
+func init() {
+	klog.InitFlags(nil)
+}
+
+var (
+	metricsAddress = flag.String("metrics-address", "0.0.0.0:29604", "export the metrics")
+)
+
+const (
+	filterRequestStr     = "filter"
+	prioritizeRequestStr = "prioritize"
+	pingRequestStr = "ping"
+)
+
+func main() {
+	flag.Parse();
+	exportMetrics()
+	klog.V(5).Infof("Starting azschedulerextender ...")
+	server := &http.Server{Addr: ":8080"}
+	http.HandleFunc("/", serveSchedulerTraffic)
+	if err := server.ListenAndServe(); err != http.ErrServerClosed {
+		klog.Fatalf("Exiting. Error starting extender server: %v", err)
+	}
+	klog.V(5).Infof("Exiting azschedulerextender ...")
+	os.Exit(0)
+}
+
+func serveSchedulerTraffic(response http.ResponseWriter, request *http.Request) {
+	if strings.Contains(request.URL.Path, filterRequestStr) {
+		filterRequest(response, request)
+	} else if strings.Contains(request.URL.Path, prioritizeRequestStr) {
+		prioritizeRequest(response, request)
+	} else if strings.Contains(request.URL.Path, pingRequestStr) {
+		pingRequest(response, request)
+	} else {
+		klog.Errorf("serveSchedulerTraffic: Not implemented request received. Request: %s . Status: %v", request.URL.Path, http.StatusNotImplemented)
+		http.Error(response, "Not implemented request received. Request: " + request.URL.Path, http.StatusNotImplemented)
+	}
+}
+
+func pingRequest(response http.ResponseWriter, request *http.Request) {
```

Comment:
Except for property getters, I find it clearer to use a Verb-Noun naming convention for functions. Consider `handlePingRequest`.